modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/FutureExecutionCallbackAdapter.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.runtime.operation;
+
+import org.mule.runtime.extension.api.runtime.operation.CompletableComponentExecutor.ExecutorCallback;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * ExecutorCallback adapter which delegates to the corresponding methods in a CompletableFuture.
+ */
+class FutureExecutionCallbackAdapter implements ExecutorCallback {
```

Comment:
Just a comment: I think it's ok to leave this a package class (preferable to an inner class) as this will be used only in this package.